### PR TITLE
Adds a sample test based on the new test framework

### PR DESCRIPTION
**Pull request by g-awmalik**

### CODE REVIEW

**Comment by comment-bot-dev** [comment]
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

File: test/integration/mysql-public/mysql_public_test.go [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+		assert.True(op.Get("settings.ipConfiguration.ipv4Enabled").Bool(), "ipv4 is enabled")
+
+		// assert values that are greater than or equal to the expected value
+		assert.GreaterOrEqual(op.Get("settings.dataDiskSizeGb").Float(), 10.0, "at least 5 backups are retained")
+
+		// assert values that are supposed to be empty or nil
+		assert.Empty(op.Get("settings.userLabels"), "no labels are set")
+
+		// assert values that are supposed to be equal to the expected values
+		assert.Equal(databaseVersion, op.Get("databaseVersion").String(), "database versions is valid is set to "+databaseVersion)
```
Comment (bharathkkb) [inline test/integration/mysql-public/mysql_public_test.go:62]:
```suggestion
		assert.Equal(databaseVersion, op.Get("databaseVersion").String(), "database versions is valid and set to "+databaseVersion)
```

File: test/integration/mysql-public/mysql_public_test.go [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+		assert.Equal(region, op.Get("region").String(), "Equals ===> GCE region is valid")
```
Comment (bharathkkb) [inline test/integration/mysql-public/mysql_public_test.go:63]:
```suggestion
		assert.Equal(region, op.Get("region").String(), "GCE region is valid")
```

File: test/integration/mysql-public/mysql_public_test.go [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+		assert.Equal(tier, op.Get("settings.tier").String(), "database tier is valid")
+		assert.Equal(true, op.Get("settings.ipConfiguration.requireSsl").Bool(), "SSL is required")
+
+		// the gjson struct (op) also allow for easy parsing through arrays and maps
+		authNetworks := op.Get("settings.ipConfiguration.authorizedNetworks").Array()
```
Comment (bharathkkb) [inline test/integration/mysql-public/mysql_public_test.go:68]: we should assert only one authorized network expected

File: test/integration/mysql-public/mysql_public_test.go [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+	const gceZone = `us-central1-c`
+	const region = `us-central1`
+	const tier = `db-n1-standard-1`
+	const authNetName = `sample-gcp-health-checkers-range`
+	const authNetCidr = `130.211.0.0/28`
+	const dbFlagName = `log_bin_trust_function_creators`
+	const dbFlagValue = `on`
+
+	mySqlT := tft.NewTFBlueprintTest(t)                   // initialize Terraform test from the Blueprints test framework
+	mySqlT.DefineVerify(func(assert *assert.Assertions) { // define and write a custom verifier for this test case
```
Comment (bharathkkb) [inline test/integration/mysql-public/mysql_public_test.go:41]: suggest we move comments above line for readability throughout
```suggestion
        // define and write a custom verifier for this test case
	mySqlT.DefineVerify(func(assert *assert.Assertions) { // define and write a custom verifier for this test case
```

File: test/integration/mysql-public/mysql_public_test.go [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+// limitations under the License.
+
+// define test package name
+package mysql_public
+
+import (
+	"fmt"
+	"testing"
+
+	// import the new Blueprints test framework modules for testing and assertions
```
Comment (bharathkkb) [inline test/integration/mysql-public/mysql_public_test.go:22]:
```suggestion
	// import the blueprints test framework modules for testing and assertions
```

File: test/integration/discover_test.go [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+package test
+
+import (
+	"testing" // should be imported to enable testing for GO modules
+
+	"github.com/GoogleCloudPlatform/cloud-foundation-toolkit/infra/blueprint-test/pkg/tft" // should be
+	// imported to enable the Blueprints helper modules for the test framework
+)
+
+func TestAll(t *testing.T) { // entry function for the test; can be named as Test*
```
Comment (bharathkkb) [inline test/integration/discover_test.go:24]: Should we also enable this in cloudbuild or are you planning that in a follow up?

File: test/integration/discover_test.go [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+	tft.AutoDiscoverAndTest(t) // the helper callback to autodiscover and test the
+	// TF examples and/or fixtures setup in the Blueprint
```
Comment (bharathkkb) [inline test/integration/discover_test.go:26]:
```suggestion
```

File: test/integration/mysql-public/mysql_public_test.go (hunk 1 above)
Comment (g-awmalik) [reply test/integration/mysql-public/mysql_public_test.go:62]: done

File: test/integration/mysql-public/mysql_public_test.go (hunk 2 above)
Comment (g-awmalik) [reply test/integration/mysql-public/mysql_public_test.go:63]: Acked

File: test/integration/mysql-public/mysql_public_test.go (hunk 3 above)
Comment (g-awmalik) [reply test/integration/mysql-public/mysql_public_test.go:68]: It is one and that's what's being asserted. It's just that it's a collection and you have to process it as as array.

File: test/integration/mysql-public/mysql_public_test.go (hunk 4 above)
Comment (g-awmalik) [reply test/integration/mysql-public/mysql_public_test.go:41]: Acked

File: test/integration/mysql-public/mysql_public_test.go (hunk 5 above)
Comment (g-awmalik) [reply test/integration/mysql-public/mysql_public_test.go:22]: done

File: test/integration/discover_test.go (hunk 6 above)
Comment (g-awmalik) [reply test/integration/discover_test.go:24]: Let's do that as a follow-up. I want to get this in so we can also publish the user guide with the right links. We can add more tests shortly after.

File: test/integration/discover_test.go (hunk 7 above)
Comment (g-awmalik) [reply test/integration/discover_test.go:26]: Done

File: build/int.cloudbuild.yaml [hunk 8]
```diff
@@ -81,6 +81,29 @@ steps:
     args: ['/bin/bash', '-c', 'source /usr/local/bin/task_helper_functions.sh && kitchen_do destroy mysql-public-local']
     timeout: 1800s
 
+  - id: go-init-mysql-public
+    waitFor:
+      - destroy-mysql-public-local
```
Comment (bharathkkb) [inline build/int.cloudbuild.yaml:86]: @morgante suggested we remove the older ruby tests as we swap these out with newer tests

File: test/integration/discover_test.go [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package test
+
+import (
+	// should be imported to enable testing for GO modules
+	"testing"
+
+	// should be imported to enable the Blueprints helper modules for the test framework
```
Comment (bharathkkb) [inline test/integration/discover_test.go:21]:
```suggestion
	// should be imported to use terraform helpers in blueprints test framework
```

File: test/integration/discover_test.go [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+	"github.com/GoogleCloudPlatform/cloud-foundation-toolkit/infra/blueprint-test/pkg/tft"
+)
+
+// entry function for the test; can be named as Test*
+func TestAll(t *testing.T) {
+	// the helper callback to autodiscover and test blueprint examples
```
Comment (bharathkkb) [inline test/integration/discover_test.go:27]:
```suggestion
	// the helper to autodiscover and test blueprint examples
```

File: test/integration/mysql-public/mysql_public_test.go [hunk 11]
```diff
@@ -0,0 +1,83 @@
...
+	const authNetCidr = `130.211.0.0/28`
+	const dbFlagName = `log_bin_trust_function_creators`
+	const dbFlagValue = `on`
+
+	// initialize Terraform test from the Blueprints test framework
+	mySqlT := tft.NewTFBlueprintTest(t)
+
+	// define and write a custom verifier for this test case call the default verify for confirming no additional changes
+	mySqlT.DefineVerify(func(assert *assert.Assertions) {
+		mySqlT.DefaultVerify(assert)
```
Comment (bharathkkb) [inline test/integration/mysql-public/mysql_public_test.go:45]:
```suggestion
                 // perform default verification ensuring Terraform reports no additional changes  on an applied blueprint
		mySqlT.DefaultVerify(assert)
```

File: test/integration/mysql-public/mysql_public_test.go (hunk 3 above)
Comment (bharathkkb) [reply test/integration/mysql-public/mysql_public_test.go:71]: IIUC there could be a list of authorized networks, shouldn't we assert that `len(authNetworks) == 1`?

**Review by bharathkkb — APPROVED** [review]
LGTM, minor comments

File: test/integration/mysql-public/mysql_public_test.go (hunk 3 above)
Comment (g-awmalik) [reply test/integration/mysql-public/mysql_public_test.go:71]: I see what you mean. I'll add that assertion.

File: build/int.cloudbuild.yaml (hunk 8 above)
Comment (g-awmalik) [reply build/int.cloudbuild.yaml:86]: Acked

File: test/integration/mysql-public/mysql_public_test.go (hunk 11 above)
Comment (g-awmalik) [reply test/integration/mysql-public/mysql_public_test.go:45]: Added

File: test/integration/discover_test.go (hunk 10 above)
Comment (g-awmalik) [reply test/integration/discover_test.go:27]: Acked

File: test/integration/discover_test.go (hunk 9 above)
Comment (g-awmalik) [reply test/integration/discover_test.go:21]: Acked

File: build/int.cloudbuild.yaml [hunk 12]
```diff
@@ -63,22 +63,27 @@ steps:
...
-    args: ['/bin/bash', '-c', 'source /usr/local/bin/task_helper_functions.sh && kitchen_do verify mysql-public-local']
-  - id: destroy-mysql-public-local
+    args: ['/bin/bash', '-c', 'cd test/integration && RUN_STAGE=verify go test -v -run TestMySqlPublicModule ./... -p 1']
+  - id: go-destroy-mysql-public
     waitFor:
-      - verify-mysql-public-local
+      - go-verify-mysql-public
     name: 'gcr.io/cloud-foundation-cicd/$_DOCKER_IMAGE_DEVELOPER_TOOLS:$_DOCKER_TAG_VERSION_DEVELOPER_TOOLS'
-    args: ['/bin/bash', '-c', 'source /usr/local/bin/task_helper_functions.sh && kitchen_do destroy mysql-public-local']
+    args: ['/bin/bash', '-c', 'cd test/integration && RUN_STAGE=destroy go test -v -run TestMySqlPublicModule ./... -p 1']
```
Comment (bharathkkb) [inline build/int.cloudbuild.yaml:86]:
```suggestion
    args: ['/bin/bash', '-c', 'cd test/integration && RUN_STAGE=teardown go test -v -run TestMySqlPublicModule ./... -p 1']
```

**Review by bharathkkb — APPROVED** [review]

**Comment by g-awmalik** [comment]
I don't seem to have access to merge this anymore. Do you see it?

**Pull request merged**